internal/components/components.go

```diff
@@ -4,29 +4,47 @@ import (
 	"context"
 
 	"github.com/grafana/grafana/pkg/schema"
+	"github.com/grafana/thema"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
-// Coremodel is an interface that must be implemented by each coremodel.
-type Coremodel interface {
+// KubeModel is an interface that must be implemented by each KubeModel-style representation of a Grafana model.
+type KubeModel interface {
```

Comment:
If we aim for clear separation of concerns, I'd move all kube-related interfaces / loaders to a subpackage, e.g. `internal/components/kube` or similar.

Reply:
yeah, agreed